build_runner/lib/src/package_graph/apply_builders.dart

```diff
@@ -189,16 +197,20 @@ class BuilderApplication {
     BuilderOptions defaultDevOptions,
     BuilderOptions defaultReleaseOptions,
   }) {
-    var phaseFactory = (String package, BuilderOptions options,
+    var phaseFactory = (PackageNode package, BuilderOptions options,
         InputSet targetSources, InputSet generateFor, bool isReleaseBuild) {
       generateFor ??= defaultGenerateFor;
 
       final optionsWithDefaults = (defaultOptions ?? BuilderOptions.empty)
           .overrideWith(
               isReleaseBuild ? defaultReleaseOptions : defaultDevOptions)
           .overrideWith(options);
+      if (package.isRoot) {
+        optionsWithDefaults.overrideWith(new BuilderOptions({}, isRoot: true));
```

Comment:
I think we need `optionsWithDefaults = optionsWithDefaults.overrideWith`...

Reply:
yep.... good catch